pkgs/by-name/ne/newlisp/package.nix

```
@@ -0,0 +1,47 @@
+{
+  lib,
+  fetchurl,
+  stdenv,
+  libffi,
+  readline,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "newlisp";
+  version = "10.7.5";
+
+  src = fetchurl {
+    url = "https://www.newlisp.org/downloads/newlisp-${finalAttrs.version}.tgz";
+    hash = "sha256-3C0P9lHCsnW8SvOvi6WYUab7bh6t3CCudftgsekBJuw=";
+  };
+
+  nativeBuildInputs = [
+    libffi
+    readline
+  ];
+
+  configureScript = "./configure-alt";
+
+  doCheck = true;
+  checkTarget = "testall";
+
+  meta = {
+    description = "Lisp-like, general-purpose scripting language";
+    longDescription = ''
+      newLISP is a Lisp-like, general-purpose scripting language. It is
+      especially well-suited for applications in AI, simulation, natural
+      language processing, big data, machine learning and statistics. Because
+      of its small resource requirements, newLISP is excellent for embedded
+      systems applications. Most of the functions you will ever need are
+      already built in. This includes networking functions, support for
+      distributed and multicore processing, and Bayesian statistics.
+    '';
+    homepage = "https://www.newlisp.org/";
+    downloadPage = "https://www.newlisp.org/downloads/";
+    changelog = "https://www.newlisp.org/downloads/newlisp-${finalAttrs.version}/doc/CHANGES";
+    license = lib.licenses.gpl3Only;
```

```suggestion
    license = lib.licenses.gpl3Plus;
```

see https://salsa.debian.org/debian/newlisp/-/blob/dcf35d3b53ce5ad5a6c1177af6f49c93de44c1f5/newlisp.h#L8

---

I see. The [manual](https://www.newlisp.org/downloads/manual_frame.html), which I put into account when committing this, only cites GPL v3. However, the [source code](https://www.newlisp.org/downloads/newlisp-10.7.5/newlisp.h) allow any later versions indeed.

---

Yeah, it is a bit tricky to find out whether it is `gpl3Only` or `gpl3Plus`.

https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206#how-to-determine-license-3
https://www.gnu.org/licenses/identify-licenses-clearly.html